compiler/cli/cli-py/src/org/jetbrains/kotlin/cli/py/K2PyCompiler.kt

```diff
@@ -217,20 +212,11 @@ class K2PyCompiler : CLICompiler<K2PyCompilerArguments>() {
             val compiledModule = compile(
                 module,
                 phaseConfig,
-                if (arguments.irDceDriven) PersistentIrFactory() else IrFactoryImpl,
```

Comment:
I was hoping we'll tackle implementing DCE (#102) soon-ish, with hope to solve the MicroPython issue (#84), but it's fine as well to go towards simplifying now, and revive the code later, perhaps in a simplified and cleaner form.